### PR TITLE
fix(sca): do not skip the scan if BC_LIC is used with --check

### PR DESCRIPTION
**Pull request by mikeurbanski1**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Fixes an issue where running with `-c BC_LIC_1` caused the SCA scans to get skipped entirely, returning no results.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my feature, policy, or fix is effective and works
- [X] New and existing tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Saarett — APPROVED** [review]
🌹

File: tests/sca_image/test_runner.py [hunk 1]
```diff
@@ -113,19 +113,12 @@ def test_runner_honors_enforcement_rules(mock_bc_integration, image_name, cached
...
     image_runner = Runner()
-    filter = RunnerFilter(framework=['sca_image'], use_enforcement_rules=True)
-    # this is not quite a true test, because the checks don't have severities. However, this shows that the check registry
-    # passes the report type properly to RunnerFilter.should_run_check, and we have tests for that method
-    filter.enforcement_rule_configs = {CheckType.SCA_IMAGE: Severities[BcSeverities.OFF]}
+    filter = RunnerFilter(framework=['sca_image'], checks=['BC_LIC_1'])
     image_runner.image_referencers = [GHA_Runner()]
     report = image_runner.run(root_folder=WORKFLOW_EXAMPLES_DIR, runner_filter=filter)
 
-    summary = report.get_summary()
```
Comment (itai1357) [inline tests/sca_image/test_runner.py:123]: what is the reason for removing this test?
is the summary tested in other places?

File: checkov/sca_package/runner.py [hunk 2]
```diff
@@ -35,8 +35,8 @@ def prepare_and_scan(
         runner_filter = runner_filter or RunnerFilter()
         excluded_file_names = excluded_file_names or set()
 
-        # skip complete run, if flag '--check' was used without a CVE check ID
-        if runner_filter.checks and all(not check.startswith("CKV_CVE") for check in runner_filter.checks):
+        # skip complete run, if flag '--check' was used without a CVE check ID or the license policies
+        if runner_filter.checks and all(not (check.startswith("CKV_CVE") or check.startswith("BC_CVE") or check.startswith("BC_LIC")) for check in runner_filter.checks):
```
Comment (itai1357) [inline checkov/sca_package/runner.py:39]: what do you say about moving this condition (which is not very short and is repeated) to a function, maybe in the file `checkov/common/sca/commons.py`

File: tests/sca_image/test_runner.py (hunk 1 above)
Comment (mikeurbanski1) [reply tests/sca_image/test_runner.py:123]: Oops, I forgot to copy and change this one 🙈

**Review by itai1357 — APPROVED** [review]

**Pull request merged**